core/src/indexer/native_transfer.rs

```diff
@@ -154,16 +160,37 @@ pub async fn native_transfer_block_consumer(
     provider: Arc<JsonRpcCachedProvider>,
     block_numbers: &[U64],
     network_name: &str,
-    config: &TraceProcessingConfig,
+    config: Arc<TraceProcessingConfig>,
 ) -> Result<(), ProcessEventError> {
     let trace_futures: Vec<_> =
-        block_numbers.iter().map(|n| provider_call(provider.clone(), config, *n)).collect();
+        block_numbers.iter().map(|n| provider_call(provider.clone(), &config, *n)).collect();
     let trace_calls = try_join_all(trace_futures).await?;
     let (from_block, to_block) =
         block_numbers.iter().fold((U64::MAX, U64::zero()), |(min, max), &num| {
             (std::cmp::min(min, num), std::cmp::max(max, num))
         });
 
+    // We're not ready to support complete "trace" indexing for zksync chains. So we can
+    // effectively only get what we need for native transfers by removing calls to "system
+    // contracts".
+    //
+    // As an example, a Zksync ETH transfer have a complex set of interactions with contracts like:
+    // - `0x0000000000000000000000000000000000008009`
+    // - `0x0000000000000000000000000000000000008001`
+    // - `0x000000000000000000000000000000000000800a`
+    //
+    // There will be one deeply-nested "transfer" call for the actual two EOAs, so filtering
+    // everything else will allow us to grab that.
+    //
+    // Read more: https://docs.zksync.io/zksync-protocol/contracts/system-contracts#l2basetoken-msgvaluesimulator
```

Comment:
Might need some input here on a better way to do this.

Reply:
i think its only `0x000000000000000000000000000000000000800A` you need to care for on the transfer of native but it does have other system contracts is your plan to exclude them?

Reply:
https://github.com/matter-labs/zksync-era/blob/7f36ed98fc6066c1224ff07c95282b647a8114fc/infrastructure/zk/src/verify-upgrade.ts#L24